pkg/winperfcounters/watcher.go

```diff
@@ -50,26 +51,34 @@ func (w Watcher) Path() string {
 	return w.Counter.Path()
 }
 
-func (w Watcher) ScrapeData() (CounterValue, error) {
-	counterValues, err := w.Counter.ScrapeData()
+func (w Watcher) ScrapeData() ([]CounterValue, error) {
+	scrapedCounterValues, err := w.Counter.ScrapeData()
 	if err != nil {
-		return CounterValue{}, err
+		return []CounterValue{}, err
 	}
-	metric := w.GetMetricRep()
 
-	if len(counterValues) != 1 {
-		return CounterValue{}, fmt.Errorf("returned incorrect amount of counter values: %d", len(counterValues))
+	if strings.Contains(w.Path(), "*") {
+		counterValues := []CounterValue{}
+		for _, counterValue := range scrapedCounterValues {
+			metric := w.GetMetricRep()
+			if counterValue.InstanceName != "" {
+				if metric.Attributes == nil {
+					metric.Attributes = map[string]string{instanceLabelName: counterValue.InstanceName}
+				}
+				metric.Attributes[instanceLabelName] = counterValue.InstanceName
+				counterValues = append(counterValues, CounterValue{MetricRep: metric, Value: counterValue.Value})
+			}
+		}
+		return counterValues, nil
```

Comment:
Updated

Reply:
Ok, looking at this closer, does it even need to worry about whether it's single vs multiple? Why not just always loop over it? If it's one, there will be one iteration. 

The only change really is that the expectation of getting one vs not one is not validated, but this seems low value and unlikely to be a problem anyways.

Reply:
Fair, the main reason I was going to leave it in there is if there was multiple values there would be data loss because there'd be two paths with the same path and no instance label to differentiate them. However, I do think you're right that it's probably unlikely

Reply:
> same path and no instance label to differentiate them

Is there a specific scenario that could cause this, or is this just in theory what would cause an issue?

You could still check if `if strings.Contains(w.Path(), "*") and len(...) > 1` and return an error, but otherwise process them the same.

Reply:
More just a theory.